src/gpsea/analysis/pscore/_impl.py

```diff
@@ -110,3 +110,178 @@ def score(
+
+        Returns:
+            1 if the term or any descendant is present in the patient, otherwise 0
+        """
+        total_count = 0
+        for term_id in observed_term_ids:
+            for desc_tid in self._hpo.graph.get_ancestors(term_id, include_source=True):
+                if desc_tid.value == target_tid:
+                    total_count += 1
+        return total_count
+
+    def _postnatal_growth_score(self, observed_term_ids: typing.List[str]) -> float:
+        """
+        calculate the postnatal growth component of the score
+        Args:
+            observed_term_ids: terms observed in patient
+
+        Returns:
+
+        """
+        microcephaly = 'HP:0000252'
+        short_stature = 'HP:0004322'
+        macrocephaly = 'HP:0000256'
+        tall_stature = 'HP:0000098'
+        total_count = 0
+        for tid in [microcephaly, short_stature, macrocephaly, tall_stature]:
+            total_count += self._term_or_descendant(tid, observed_term_ids)
+        if total_count > 2:
+            raise ValueError(f"Inconsistent annotations for postnatal growth score {total_count}:  {observed_term_ids}")
+        return total_count
+
+    def _facial_dysmorphism_score(self, observed_term_ids: typing.List[str]) -> float:
+        """
+        This section assigns two points if two or more anomalies are identified in the following
+        categories: hypertelorism, nasal anomalies and ear anomalies. Our implementation counts the total
+        number of terms or descendents of the hypertelorism, Abnormal external nose morphology, and
+        Abnormal pinna morphology.
+
+        Args:
+            observed_term_ids: terms observed in patient
+
+        Returns: facial dysmorphism score (between 0 and 2)
+
+        """
+        hypertelorism = 'HP:0000316'
+        external_nose = 'HP:0010938'
+        pinna_morphology = 'HP:0000377'
+        total_count = len([t for t in observed_term_ids if t == hypertelorism])
+        total_count += self._term_or_descendant_count(target_tid=external_nose, observed_term_ids=observed_term_ids)
+        total_count += self._term_or_descendant_count(target_tid=pinna_morphology, observed_term_ids=observed_term_ids)
+        if total_count > 1:
+            return 2
+        else:
+            return 0
+
+    def _congenital_score(self, observed_term_ids: typing.List[str]) -> float:
+        """
+        Non-facial dysmorphism and congenital abnormalities component
+        One point is assigned for either the corresponding HPO terms or any of their descendents up to a maximum of 2.
+        Args:
+            observed_term_ids:  terms observed in patient
+
+        Returns:   Non-facial dysmorphism and congenital abnormalities score (between 0 and 2)
+
+        """
+        hypospadias = 'HP:0000047'
+        abnormal_hand_morphology = 'HP:0005922'
+        abnormal_heart_morphology = 'HP:0001627'
+        total_count = len([t for t in observed_term_ids if t == hypospadias])
```

Comment:
Same as above, this code does not check Hypospadias descendants.

I will fix the code to check the descendants. @pnrobinson please should if it should not be done!

Reply:
see above, hypertelorism does not have descendats

Reply:
I am afraid I got you on this one.. https://hpo.jax.org/browse/term/HP:0000047

Hypospadias actually *does* have descendants.